modules/compute/vm-instance/main.tf

```diff
@@ -45,6 +45,18 @@ locals {
     "ENABLE"  = "TRUE"
   }
   enable_oslogin = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+
+  machine_vals            = split("-", var.machine_type)
+  machine_family          = local.machine_vals[0]
+  machine_not_shared_core = length(local.machine_vals) > 2
+  machine_vcpus           = try(parseint(local.machine_vals[2], 10), 1)
+
+  smt_capable_family = !contains(["t2d"], local.machine_family)
+  smt_capable_vcpu   = local.machine_vcpus >= 2
+
+  smt_capable = local.smt_capable_family && local.smt_capable_vcpu && local.machine_not_shared_core
```

Comment:
nit: I don't think you need to define and use `machine_not_shared_core` since its value will always match that of `smt_capable_vcpu`. For me, it adds extra complexity, but I would also understand if you want to keep it since the behavior is kind of relying on the fallback value of `machine_vcpus`.

Reply:
ACK. 
I added it based on https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/382#discussion_r911420156 
I prefer keeping `machine_not_shared_core` explicit for code readability/maintainability.